libr/anal/types.c

```diff
@@ -299,15 +299,29 @@ R_API char *r_anal_type_func_args_name(RAnal *anal, const char *func_name, int i
 
 #define MIN_MATCH_LEN 4
 
-// TODO: Future optimizations
-// - symbol names are long and noisy, reduce searchspace for match here
-//   by preprocessing to delete noise, as much as we can.
-// - We ignore all func.*, but thats maybe most of sdb entries
-//   could maybe eliminate this work (strcmp).
+static char *type_func_try_guess(RAnal *anal, char *name) {
+	if (strlen (name) < MIN_MATCH_LEN) {
+		return NULL;
+	}
+	if (sdb_exists (anal->sdb_types, name)) {
+		const char *res = sdb_const_get (anal->sdb_types, name, 0);
```

Comment:
if sdb_const_get returns NULL, you can assume that sdb_exists, so you can save 1 sdb call here.